langchain4j-core/src/main/java/dev/langchain4j/data/document/DocumentLoader.java

```diff
@@ -5,10 +5,7 @@
 /**
  * Utility class for loading documents.
  */
-public class DocumentLoader {
-
-    private DocumentLoader() {
-    }
+public interface DocumentLoader {
```

Comment:
Why is this change required?

Reply:
Because there is no interface for DocumentLoaders and now e.g. `FileSystemDocumentLoader` or`AmazonS3DocumentLoader` can implement the at least marker interface.
It would be even better to implement :

```java
interface DocumentLoader<T extends Document> {
   T loadDocument(DocumentSource<T> source);
}
```

Spring AI has a similar concept of [DocumentReader](https://github.com/spring-projects/spring-ai/blob/main/spring-ai-core/src/main/java/org/springframework/ai/document/DocumentReader.java)

Reply:
@kpavlov loaders were left out without a common interface on purpose, as they all have different APIs and it is hard to find a common one without forcing a lot of limitations. Also loaders usually are used in the beginning of the pipeline, so it is not really required for them to be polymorphic.

I am OK with a marker interface solely for discoverability, but I would not force a common API.
BTW, users are not supposed to use `DocumentSource` directly, they are supposed to use `XyzDocumentLoader` which will then deal with `DocumentSource`s internally.